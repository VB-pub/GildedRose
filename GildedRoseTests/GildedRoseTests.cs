using GildedRoseKata.Extensions;
using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using GildedRose.Models;

namespace GildedRoseTests
{
    [TestClass]
    public partial class GildedRoseTests
    {
        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void UpdateQualityTest()
        {
            CombinationApprovals.VerifyAllCombinations(StartUpdateQuality,
                _names, _sellIns, _qualities);
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void UpdateConjuredQualityTest()
        {
            CombinationApprovals.VerifyAllCombinations(StartUpdateQuality,
                _namesConjured, _sellIns, _qualities);
        }

        private string StartUpdateQuality(string name, int sellIn, int quality)
        {
            Item[] items = new Item[] {
                
                new Item
                {
                    Name = name,
                    SellIn = sellIn,
                    Quality = quality
                }
            };

            GildedRoseKata.GildedRose app = new(items);
            app.UpdateQuality();
            return app.Items[0].ConvertToString(delimiter);
        }
    }
}