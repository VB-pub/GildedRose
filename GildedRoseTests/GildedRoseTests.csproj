<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>

    <IsPackable>false</IsPackable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="ApprovalTests" Version="5.7.3" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
    <PackageReference Include="MSTest.TestAdapter" Version="2.2.8" />
    <PackageReference Include="MSTest.TestFramework" Version="2.2.8" />
    <PackageReference Include="coverlet.collector" Version="3.1.2" />
  </ItemGroup>

	<ItemGroup>
		<ProjectCapability Include="ConfigurableFileNesting" />
		<ProjectCapability Include="ConfigurableFileNestingFeatureEnabled" />
	</ItemGroup>

	<ItemGroup>
	  <ProjectReference Include="..\GildedRose\GildedRose.csproj" />
	</ItemGroup>

	<ItemGroup>
	  <None Update="GildedRoseTests.UpdateQualityTest.approved.txt">
	    <DependentUpon>GildedRoseTests.cs</DependentUpon>
	  </None>
	</ItemGroup>

</Project>
