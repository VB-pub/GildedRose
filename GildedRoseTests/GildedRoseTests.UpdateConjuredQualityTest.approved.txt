[Conjured foo,-1,0] => Conjured foo;-2;0
[Conjured foo,-1,1] => Conjured foo;-2;0
[Conjured foo,-1,49] => Conjured foo;-2;45
[Conjured foo,-1,50] => Conjured foo;-2;46
[Conjured foo,0,0] => Conjured foo;-1;0
[Conjured foo,0,1] => Conjured foo;-1;0
[Conjured foo,0,49] => Conjured foo;-1;45
[Conjured foo,0,50] => Conjured foo;-1;46
[Conjured foo,5,0] => Conjured foo;4;0
[Conjured foo,5,1] => Conjured foo;4;0
[Conjured foo,5,49] => Conjured foo;4;47
[Conjured foo,5,50] => Conjured foo;4;48
[Conjured foo,10,0] => Conjured foo;9;0
[Conjured foo,10,1] => Conjured foo;9;0
[Conjured foo,10,49] => Conjured foo;9;47
[Conjured foo,10,50] => Conjured foo;9;48
[Conjured foo,11,0] => Conjured foo;10;0
[Conjured foo,11,1] => Conjured foo;10;0
[Conjured foo,11,49] => Conjured foo;10;47
[Conjured foo,11,50] => Conjured foo;10;48
